src/Nethermind/Nethermind.Evm/Precompiles/AddressExtensions.cs

```diff
@@ -2,47 +2,41 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 using System;
+using System.Runtime.InteropServices;
 using Nethermind.Core;
-using Nethermind.Core.Extensions;
 using Nethermind.Core.Specs;
 
-namespace Nethermind.Evm.Precompiles
+namespace Nethermind.Evm.Precompiles;
+
+public static class AddressExtensions
 {
-    public static class AddressExtensions
+    public static bool IsPrecompile(this Address address, IReleaseSpec releaseSpec)
     {
-        private static byte[] _nineteenZeros = new byte[19];
-
-        public static bool IsPrecompile(this Address address, IReleaseSpec releaseSpec)
-        {
-            if (!Bytes.AreEqual(address.Bytes.AsSpan(0, 19), _nineteenZeros))
-            {
-                return false;
-            }
-
-            int precompileCode = address[19];
-            return precompileCode switch
+        Span<uint> data = MemoryMarshal.Cast<byte, uint>(address.Bytes.AsSpan());
+        return (data[4] & 0x00ffffff) == 0
+            && data[3] == 0 && data[2] == 0 && data[1] == 0 && data[0] == 0
```

Comment:
Loading to a Vector256 wouldn't be even faster?

Reply:
Anything but 4B integers does seem to work well with 20B address. May be we could cast it additionally to 8B int / Vector128(?) and have less comparisons, but it'd become too complicated